invenio_rdm_records/records/systemfields/tombstone.py

```diff
@@ -0,0 +1,192 @@
+
+    @property
+    def note(self):
+        """Get the public note attached to the tombstone."""
+        return self._note
+
+    @note.setter
+    def note(self, value):
+        """Set the public note attached to the tombstone."""
+        self._note = value or ""
+
+    @property
+    def removed_by(self):
+        """Get a reference to the entity who removed the record."""
+        return self._removed_by
+
+    @removed_by.setter
+    def removed_by(self, value):
+        """Set a reference to the entity who removed the record.
+
+        If the value is a string or integer, it is assumed that the
+        reference is a user and the value will be turned into a dictionary
+        of the shape ``{"user": value}``.
+        If the value is a referenceable entity (such as a ``User`` object),
+        the ``ResolverRegistry`` will be used to create a reference dict.
+        """
+        if value is None:
+            self._removed_by = None
+
+        elif (ref := ResolverRegistry.reference_entity(value)) is not None:
+            self._removed_by = ref
+
+        elif isinstance(value, (int, str)):
+            self._removed_by = {"user": str(value)}
+
+        else:
+            raise ValueError(f"Invalid value for 'tombstone.removed_by': {value}")
+
+    @property
+    def removal_date(self):
+        """Get the removal date."""
+        return self._removal_date
+
+    @removal_date.setter
+    def removal_date(self, value):
+        """Set the removal date."""
+        if value is None:
+            value = datetime.utcnow()
+
+        if isinstance(value, datetime):
+            value = value.isoformat()
+
+        self._removal_date = value
+
+    @property
+    def citation_text(self):
+        """Get the citation text of the tombstoned record."""
+        return self._citation_text
+
+    @citation_text.setter
+    def citation_text(self, value):
+        """Set the citation text of the tombstoned record."""
+        self._citation_text = value or ""
+
+    @property
+    def is_visible(self):
+        """Check if the tombstone is publicly visible."""
+        return self._is_visible
+
+    @is_visible.setter
+    def is_visible(self, value):
+        """Set the public visibility of the tombstone."""
+        self._is_visible = bool(value)
+
+    @property
+    def removed_by_proxy(self):
+        """Resolve the entity proxy for ``self.removed_by``."""
+        if self.removed_by is None:
+            return None
+        else:
+            return ResolverRegistry.resolve_entity_proxy(self.removed_by)
+
+    def dump(self):
+        """Dump the tombstone into a dictionary."""
+        data = {
+            "note": self.note,
+            "removed_by": self.removed_by,
+            "removal_date": self.removal_date,
+            "citation_text": self.citation_text,
+            "is_visible": self.is_visible,
+        }
+
+        if self.removal_reason:
+            data["removal_reason"] = self.removal_reason
```

Comment:
If the `removal_reason` isn't set to anything, it's better to not have the field at all.
If it is set (e.g. to `removal_reason: None`, or `removal_reason: {"id": None}`) then the `relations` field will start complaining and prevent `record.commit()` from finishing.
Because the `relations` systemfield has the checks in `pre_commit()` and lexicographically comes before `tombstone`, the latter will have no chance of updating the record's data in `pre_commit()` to rectify the wrong data.